engines/stark/services/gamemessage.cpp

```diff
@@ -0,0 +1,105 @@
+/* ResidualVM - A 3D game interpreter
+ *
+ * ResidualVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the AUTHORS
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "engines/stark/services/gamemessage.h"
+
+#include "common/file.h"
+#include "common/ini-file.h"
+
+#include "gui/message.h"
+
+namespace Stark {
+
+GameMessage::GameMessage() :
+		_texts() {
+	Common::File tmp;
+	if (!tmp.open("language.ini")) {
+		return;
+	}
+
+	// Skip the head of the language.ini that is in an unsupported format
+	Common::String line;
+	while (!tmp.eos() && !tmp.err()) {
+		line = tmp.readLine();
+		if (line.size() > 0 && line[0] == '-' && line[1] == '-') {
+			break;
+		}
+	}
+
+	Common::INIFile file;
+	if(file.loadFromStream(tmp)) {
+		Common::String section = "Language";
+		Common::String key, text;
+
+		for (uint i = 2; i <= 400; ++i) {
+			key = Common::String::format("%03d", i);
+			if (file.hasKey(key, section)) {
+				file.getKey(key, section, text);
+				_texts[i] = text;
+			}
+		}
+	}
+
+	// Pre-process some of the texts
+	if (_texts.contains(kYes)) {
+		_texts[kYes].deleteChar(0);
+	}
+	if (_texts.contains(kNo)) {
+		_texts[kNo].deleteChar(0);
+	}
+	if (_texts.contains(kOverwriteSave)) {
+		Common::replace(_texts[kOverwriteSave], "\\n", "\n");
+	}
+}
+
+bool GameMessage::alert(const Common::String &msg, TextKey leftBtnMsg, TextKey rightBtnMsg) {
+	// TODO: Build the original game's confirmation dialog
+	GUI::MessageDialog alert(msg, getTextByKey(leftBtnMsg).c_str(), getTextByKey(rightBtnMsg).c_str());
```

Comment:
Having the display code here breaks the convention where `Service` classes are for data access / backend things. Perhaps it would be better if the display methods were in `UserInterface`.